qdrant_client/qdrant_fastembed.py

```diff
@@ -819,3 +774,145 @@ def query_batch(
+
+        if query is None:
+            return None
+
+        raise ValueError(f"Unsupported query type: {type(query)}")
+
+    def _resolve_query_request(self, query: models.QueryRequest) -> models.QueryRequest:
+        query = deepcopy(query)
+        query.query = self._resolve_query(query.query)
+        return query
+
+    def _resolve_query_batch_request(
+        self, requests: Sequence[models.QueryRequest]
+    ) -> Sequence[models.QueryRequest]:
+        return [self._resolve_query_request(query) for query in requests]
+
+    def _embed_models(
+        self,
+        model: BaseModel,
+        paths: Optional[List[Path]] = None,
+        is_query: bool = False,
+    ) -> Union[BaseModel, Union[List[float], models.SparseVector]]:
+        if paths is None:
+            if isinstance(model, models.Document):
+                return self._embed_raw_data(model, is_query=is_query)
+            model = deepcopy(model)
+            paths = self._embed_inspector.inspect(model)
+        for path in paths:
+            list_model = [model] if not isinstance(model, list) else model
+            for item in list_model:
+                current_model = getattr(item, path.current, None)
+                if current_model is None:
+                    continue
+                if path.tail:
+                    self._embed_models(current_model, path.tail, is_query=is_query)
+                else:
+                    was_list = isinstance(current_model, list)
+                    current_model = (
+                        [current_model] if not isinstance(current_model, list) else current_model
+                    )
+                    embeddings = [
+                        self._embed_raw_data(data, is_query=is_query) for data in current_model
+                    ]
+                    if was_list:
+                        setattr(item, path.current, embeddings)
+                    else:
+                        setattr(item, path.current, embeddings[0])
+        return model
+
+    def _embed_raw_data(
+        self,
+        data: models.Document,
+        is_query: bool = False,
+    ) -> Union[
+        Dict[str, Union[List[float], models.SparseVector]], List[float], models.SparseVector
+    ]:
+        if isinstance(data, models.Document):
+            return self._embed_document(data, is_query=is_query)
+        elif isinstance(data, dict):
+            return {
+                key: self._embed_raw_data(value, is_query=is_query) for key, value in data.items()
+            }
+        elif isinstance(data, list):
+            # we don't want to iterate over a vector
+            if data and isinstance(data[0], float):
+                return data
+            return [self._embed_raw_data(value, is_query=is_query) for value in data]
+        return data
+
+    def _embed_document(
+        self, document: models.Document, is_query: bool = False
+    ) -> Union[List[float], models.SparseVector]:
```

Comment:
Late interaction can also return `List[List[float]]` I propose to define a type alias for this, otherwise too verbose